cpp/src/cider/benchmarks/HashMapBenchmark.cpp

```diff
@@ -0,0 +1,165 @@
+  std::mt19937 rng(std::random_device{}());  // NOLINT
+  switch (pattern) {
+    case GeneratePattern::Sequence:
+      for (auto i = 0; i < row_num; ++i) {
+        null_data[i] = Random::oneIn(null_chance, rng) ? (col_data[i] = N_MIN, true)
+                                                       : (col_data[i] = i, false);
+      }
+      break;
+    case GeneratePattern::Random:
+      if (std::is_integral<T>::value) {
+        for (auto i = 0; i < col_data.size(); ++i) {
+          null_data[i] = Random::oneIn(null_chance, rng)
+                             ? (col_data[i] = N_MIN, true)
+                             : (col_data[i] = static_cast<T>(
+                                    Random::randInt64(value_min, value_max, rng)),
+                                false);
+        }
+      } else if (std::is_floating_point<T>::value) {
+        for (auto i = 0; i < col_data.size(); ++i) {
+          null_data[i] = Random::oneIn(null_chance, rng)
+                             ? (col_data[i] = N_MIN, true)
+                             : (col_data[i] = static_cast<T>(
+                                    Random::randFloat(value_min, value_max, rng)),
+                                false);
+        }
+      } else {
+        std::string str = "Unexpected type:";
+        str.append(typeid(T).name()).append(", could not generate data.");
+        CIDER_THROW(CiderCompileException, str);
+      }
+      break;
+  }
+  return std::make_tuple(col_data, null_data);
+}
+
+// Benchmark for look up
+using BaselineLookup = HashMap<uint16_t, uint8_t, HashCRC32<uint16_t>>;
+using OptimizedLookup = FixedHashMap<uint16_t, uint8_t>;
+
+template <typename KeyType, typename HashTableType>
+static void BM_Lookup(benchmark::State& state) {
+  size_t row_num = 100;
+  KeyType value_min = std::numeric_limits<KeyType>::min();
+  KeyType value_max = std::numeric_limits<KeyType>::max();
+  size_t null_chance = 0;
+  GeneratePattern pattern = GeneratePattern::Random;
+  // GeneratePattern pattern = GeneratePattern::Sequence;
+  // state.PauseTiming();
+  std::vector<KeyType> data(row_num);
+  std::vector<bool> data_null;
+  {
+    std::tie(data, data_null) =
+        value_min > value_max
+            ? generateAndFillVector<KeyType>(row_num, pattern, null_chance)
+            : generateAndFillVector<KeyType>(
+                  row_num, pattern, null_chance, value_min, value_max);
+  }
+  // state.ResumeTiming();
+  for (auto _ : state) {
+    auto collisions = bench<KeyType, HashTableType>(data, "Benchmark Lookup");
+    state.counters["Collisions"] = collisions;
+  }
+}
+
+template <typename KeyType>
+static void BM_Baseline_Lookup(benchmark::State& state) {
+  BM_Lookup<KeyType, BaselineLookup>(state);
+}
+
+template <typename KeyType>
+static void BM_Optimized_Lookup(benchmark::State& state) {
+  BM_Lookup<KeyType, OptimizedLookup>(state);
+}
+
+BENCHMARK(BM_Baseline_Lookup<uint8_t>)->RangeMultiplier(10)->Range(10, 1000);

```

Comment:
seems range are not in use?

Reply:
`Range` takes effect since there are `10, 100, 1000` three lines for each case as the screenshot of PR description shows above.

Reply:
you should use it like `auto loop_length = state.range(0)` in your benchmark function body.
example: https://github.com/google/benchmark/blob/main/docs/user_guide.md#passing-arguments

Reply:
and the screenshot shows no difference for 10/100/1000, right?

Reply:
Updated.